site/en/docs/extensions/reference/userscripts/index.md

```diff
@@ -0,0 +1,60 @@
+---
+api: userScripts
+---
+
+## Manifest
+
+To use the `chrome.userScripts` API, add the `"userScripts"` permission to your manifest.json and `"host_permissions"` for sites you want to run scripts on.
+
+```json/9-15
+{
+  "name": "User script test extension",
+  "manifest_version": 3,
+  "minimum_chrome_version": "118",
+  "permissions": [
+    "userScripts"
+  ],
+  "host_permissions": [
+    "*://requested.com/*"
+  ]
+}
+```
+
+## Concepts and usage
+
+A user script is a bit of code injected into a web page to modify its appearance or behavior. Scripts are either created by users or downloaded from a script repository using a type of extension called a user script manager.
+
+### Developer mode for extension users
+
+As an extension developer, you already have Developer mode enabled in your installation of Chrome. For user script extensions, your users will also need to enable developer mode. Here are instructions that you can copy and paste into your own documentation.
```

Comment:
Thanks for this! Can we also add a note on how to detect if developer mode is enabled? We can use the following from the [sample](https://github.com/GoogleChrome/chrome-extensions-samples/pull/1023):

```
/**
 * Checks if the user has developer mode enabled, which is required to use the
 * User Scripts API.
 *
 * @returns If the chrome.userScripts API is available.
 */
function isUserScriptsAvailable() {
  try {
    // Property access which throws if developer mode is not enabled.
    chrome.userScripts;
    return true;
  } catch {
    // Not available.
    return false;
  }
}
```

Reply:
Why is this not:

```javascript
if (userScripts in chrome) { ... }
```

Reply:
Unfortunately that will always return true if you have the `userScripts` permission. The only way to detect developer mode is to try to actually access the property which will throw an error.